src/components/firebase-support.ts

```diff
@@ -0,0 +1,57 @@
+import firebase from "firebase/compat/app";
+import "firebase/compat/messaging";
+import { Client, PushNotification } from "@twilio/conversations";
+
+export const initFcmServiceWorker = async () => {
+  firebase.initializeApp((window as any).firebaseConfig);
+
+  navigator.serviceWorker.register("firebase-messaging-sw.js").then(
+    function (registration: ServiceWorkerRegistration) {
+      console.log("ServiceWorker registered with scope:", registration.scope);
+    },
+    function (err) {
+      console.log("ServiceWorker registration failed:", err);
+    }
+  );
+};
+
+export const subscribeFcmNotifications = (convoClient: Client) => {
+  Notification.requestPermission()
+    .then(() => {
```

Comment:
`async await` js syntax'd look more readable in this case

Reply:
Makes sense to use async/await everywhere, also annotate async functions as such.

Reply:
Fixed